artifact-version-diff/src/main/java/io/quarkus/qe/PrepareOperation.java

```diff
@@ -61,10 +68,20 @@ public static void executeProcess(List<String> command, String errorMsg, Path pa
                     .redirectOutput(ProcessBuilder.Redirect.DISCARD)
                     .redirectError(ProcessBuilder.Redirect.DISCARD)
                     .start();
+            StringBuilder output = new StringBuilder();
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    output.append(line).append("\n");
+                }
+            }
             process.waitFor();
+            if(process.exitValue()!=0) {
+                throw new RuntimeException(errorMsg + " Output: " + output);
+            }
             assertEquals(0, process.exitValue(), errorMsg);
         } catch (IOException | InterruptedException e) {
-            throw new RuntimeException(e);
+            throw new RuntimeException("Error during execution: " + errorMsg + ". Exception: " + e.getMessage());
```

Comment:
Wouldn't be usage of https://docs.oracle.com/en/java/javase/21/docs/api/java.management/javax/management/RuntimeOperationsException.html#%3Cinit%3E(java.lang.RuntimeException,java.lang.String) better to propagate full stacktrace?